lib/parser.rb

```diff
@@ -1,21 +1,31 @@
 require 'json'
 
 class Parser
-  def initialize(file)
-    @file = file
-    raise "File doesn't exist" unless File.exists?(@file)
+  def initialize(filename)
+    @filename = filename
+    raise "File doesn't exist" unless File.exist?(@filename)
   end
 
   def print_first_line
-    file = File.open(@file)
-    first_line = file.readline
-    file.close
+    filename = File.open(@filename)
+    first_line = filename.readline
+    filename.close
     first_line
   end
 
   def print_info
-    file = File.foreach(@file)
-    file_lines = file.count
-    JSON.pretty_generate({@file => {lines: file_lines}}) 
+    filename = File.foreach(@filename)
```

Comment:
A nomeclatura escolhida para a variável não condiz com o que vai ser armazenado nela. É bom se atentar a isso.
```suggestion
    file_content = File.foreach(@filename)
```